rae_hw/src/peripherals/speakers.cpp

```diff
@@ -75,19 +89,21 @@ void SpeakersNode::play_mp3(const char* mp3_file) {
 
     // Set ALSA parameters
 
-    snd_pcm_set_params(alsaHandle, SND_PCM_FORMAT_S16_LE, SND_PCM_ACCESS_RW_INTERLEAVED, 1, rate * channels, 2, 50000);
+    
+
+    snd_pcm_set_params(alsaHandle, SND_PCM_FORMAT_S16_LE, SND_PCM_ACCESS_RW_INTERLEAVED, channels, rate, 2, 50000);
```

Comment:
Won't this cause issues?

Reply:
Issue was bit further down the code in:

if(snd_pcm_writei(alsaHandle, buffer, buffer_size/2 < 0) where I was sending too big chunk of a buffer. To compensate for that I only let alsa run in 1 channel. I further divided buffer into individual bytes when sending to device and now it runs without issues in both stereo and mono files.